### PR TITLE
Replace PCRE with native regex

### PR DESCRIPTION
**Pull request by dylanjm**
This PR replaces the dependency of PCRE with c++ regex library.

refs #15081
 @dschwen

### CODE REVIEW

File: framework/src/parser/CommandLine.C [hunk 1]
```diff
@@ -92,6 +110,9 @@ CommandLine::initForMultiApp(const std::string & subapp_full_name)
       });
   _argv.erase(new_end, _argv.end());
 
+  for(size_t i = 0; i < _argv.size(); i++)
+    std::cout << _argv[i] << "\n";
```
Comment (dschwen) [inline framework/src/parser/CommandLine.C:114]: This is left from debgugging?

**Comment by permcody** [comment]
BTW - I think there is a fairly good chance that this will require a bump in our minimum compilers. Regex support has been terrible in supposedly C++11 compliant compilers. I guess we'll see.

**Comment by permcody** [comment]
OK - as I suspected GCC 4.8.4 does not support a working regex implementation. It's going to be a bit of work, but you'll need to work with Jason to figure out the new minimum compiler. I'm not going to hold us to the current minimum. We've been here a long time. Hopefully we don't have to jump too many versions ahead though.

**Comment by dschwen** [comment]
Proper `<regex>` support was added in [GCC 4.9](https://isocpp.org/blog/2014/04/gcc-4.9.0).

**Comment by permcody** [comment]
@milljm - What's the oldest GCC compiler we can get through Conda, and if it's not 4.9, any thoughts on what we can do to support a min compiler package now and moving forward?

**Comment by milljm** [comment]
The current GCC compilers available on conda:

```pre
libgcc                         4.8.4               1  conda-forge         
libgcc                         4.8.5               2  conda-forge         
libgcc                         5.2.0               0  conda-forge         
libgcc                         7.2.0      h69d50b8_2  conda-forge         
libgcc                         7.2.0      h69d50b8_2  pkgs/main 
```

I would suggest we continue to call our minimum a 'special' moose-environment package only thing.

**Comment by milljm** [comment]
Oh... you want to bump the min (I didn't bother reading all the comments). That may be problematic, but I can look into building a new stack using 5.2.0.

**Comment by dylanjm** [comment]
@dschwen somehow this file:

```
framework/contrib/exodiff/add_to_log.C
```

Got added to my commit, but I don't recall editing it. Do you know the command to remove this from my commit?

I'm expecting the tests to pass now.

**Comment by dylanjm** [comment]
@dschwen pushed a new commit to handle json cache capability. WIP since there are probably changes to apply to how we cache.

**Comment by dylanjm** [comment]
@dschwen link to json cache: https://github.com/idaholab/moose/pull/15104/files#diff-bc6d06c770db9564a1140c39d865cfe6R755

**Comment by milljm** [comment]
The minimum GCC compiler now in use is 5.1. Just stating this, as this was accomplished a few weeks ago on the build machines.

**Pull request closed without merging**